app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,47 @@
+class GroupsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_group, only: [:show, :edit, :update, :destroy]
```

Comment:
My personal preference is to avoid callbacks like this, largely because I am easily confused as to the source of `@group`, but it's pretty minor and since it's at the top of the file, fine with me if you prefer it.